classes/Repository/PaypalAccountRepository.php

```diff
@@ -184,13 +184,167 @@ public function getPaypalPaymentStatus()
+     * @return bool
+     */
+    public function isVenmoEnabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_VENMO_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isSepaLastschriftEnabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_SEPA_LASTSCHRIFT_CREDIT_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isBancontactEnabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_BANCONTACT_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEpsEnabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_EPS_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isGiropayEnabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_GIROPAY_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isIdealEnabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_IDEAL_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isMyBankEnabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_MYBANK_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isPrzelewy24Enabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_PRZELEWY24_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
+
+    /**
+     * @return bool
+     */
+    public function isSofortEnabled()
+    {
+        return (bool) \Configuration::get(
+            PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_SOFORT_ENABLED,
+            null,
+            null,
+            (int) \Context::getContext()->shop->id,
+            true
+        );
+    }
```

Comment:
The idea of having one method and constant for each payment method is good, so we can for instance enjoy autocompletion with the IDE, but there is duplicated we can avoid.

We could have a generic method:

```php
    /**
     * Generic method checking a payment method is enabled
     *
     * @return bool
     */
    public function isPaymentMethodEnabled($paymentMethod)
    {
        return (bool) \Configuration::get(
            $paymentMethod,
            null,
            null,
            (int) \Context::getContext()->shop->id,
            true
        );
    }
```

Then you can reuse it on all the other methods. For instance:

```php
   /**
     * @return bool
     */
    public function isSofortEnabled()
    {
        return $this->isPaymentMethodEnabled(PaypalAccount::PS_CHECKOUT_FUNDING_SOURCE_SOFORT_ENABLED);
    }
```

This allows you to fix only one method in the future if the way of checking a payment method evolves.